### PR TITLE
Make company_contact.contactable_by_* columns nullable and remove from the API

### PR DESCRIPTION
**Pull request by elcct**
### Description of change

This removes deprecated `contact.contactable_by_*` columns from the API and make them nullable.

Next PR should remove those fields from the database.

### Checklist

* [X] Has a new newsfragment been created? Check [changelog/README.rst](https://github.com/uktrade/data-hub-leeloo/blob/master/changelog/README.rst) for instructions
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

**Comment by reupen** [comment]
We should remove the columns from the state as well (e.g. https://github.com/uktrade/data-hub-leeloo/pull/1063/files)

File: datahub/company/serializers.py [hunk 1]
```diff
@@ -216,16 +211,6 @@ class Meta:
         validators = [
             NotArchivedValidator(),
             RulesBasedValidator(
-                ValidationRule(
-                    'contact_preferences_required',
```
Comment (reupen) [inline datahub/company/serializers.py]: Should be able to get rid of the message (for `contact_preferences_required`) as well.

File: changelog/contact/contact-contactable-by.db [hunk 2]
```diff
@@ -0,0 +1,5 @@
+The column ```contact.contactable_by_dit``` was made nullable in preparation for its removal.
```
Comment (marcofucci) [inline changelog/contact/contact-contactable-by.db]: The table is called `company_contact` I think.

Comment (elcct) [reply changelog/contact/contact-contactable-by.db]: Good catch. Corrected. Thanks.

**Pull request merged**